### PR TITLE
Add watcher-operator to openstack-k8s-operators

### PR DESCRIPTION
**Pull request by cescgina**
Initial configuration files for the watcher-operator.

### CODE REVIEW

**Review by kopecmartin — APPROVED** [review]
lgtm

**Comment by amoralej** [comment]
I should be in the openshift org now.

**Review by marios — COMMENTED** [review]
lgtm/didn't spot something

File: ci-operator/config/openstack-k8s-operators/watcher-operator/openstack-k8s-operators-watcher-operator-main.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+tests:
+- as: precommit-check
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAGS=
+    export PRE_COMMIT_HOME=/tmp
+    export GOLANGCI_LINT_CACHE=/tmp
+    mkdir -p ../operator/ && cp -r . ../operator
+    cd ../operator
+    pre-commit run -a --show-diff-on-failure
```
Comment (SeanMooney) [inline ci-operator/config/openstack-k8s-operators/watcher-operator/openstack-k8s-operators-watcher-operator-main.yaml:34]: ok so this cant pass until we merge https://github.com/openstack-k8s-operators/watcher-operator/pull/4

so ill manually checkout that pr and run pre-commit and confirm it passes.
if it does we can merge that pr then proceed with this one.

File: core-services/prow/02_config/openstack-k8s-operators/watcher-operator/_prowconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+tide:
+  queries:
```
[the remark this answers is not in the document]
Comment (cescgina) [reply core-services/prow/02_config/openstack-k8s-operators/watcher-operator/_prowconfig.yaml:4]: good point, added it

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-watcher-operator-main-precommit-check

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-watcher-operator-main-images

**Review by marios — APPROVED** [review]
both new jobs green on last pass

File: ci-operator/config/openstack-k8s-operators/watcher-operator/openstack-k8s-operators-watcher-operator-main.yaml (hunk 1 above)
Comment (marios) [reply ci-operator/config/openstack-k8s-operators/watcher-operator/openstack-k8s-operators-watcher-operator-main.yaml:34]: (that merged now and job is green here, adding lgtm label)

**Pull request merged**